textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/join/SimilarityJoinPredicate.java

```diff
@@ -0,0 +1,195 @@
+package edu.uci.ics.textdb.dataflow.join;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import edu.uci.ics.textdb.api.common.Attribute;
+import edu.uci.ics.textdb.api.common.FieldType;
+import edu.uci.ics.textdb.api.common.IField;
+import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.api.common.Schema;
+import edu.uci.ics.textdb.common.constants.SchemaConstants;
+import edu.uci.ics.textdb.common.exception.DataFlowException;
+import edu.uci.ics.textdb.common.field.DataTuple;
+import edu.uci.ics.textdb.common.field.IDField;
+import edu.uci.ics.textdb.common.field.ListField;
+import edu.uci.ics.textdb.common.field.Span;
+import edu.uci.ics.textdb.dataflow.common.IJoinPredicate;
+import info.debatty.java.stringsimilarity.NormalizedLevenshtein;
+
+public class SimilarityJoinPredicate implements IJoinPredicate {
+    
+    public static final String INNER_PREFIX = "inner_";
+    public static final String OUTER_PREFIX = "outer_";
+    
+    Double similarity;
+    String joinAttributeName;
+    
+    /**
+     * The output schema of Similarity Join will be the combination of inner and outer schema.
+     *   (except the _id, spanList and payload field)
+     *   
+     * Example of a same-table, different-tuple join
+     * table_schema,   inner_tuple,             outer_tuple
+     *   _id:          random_id                random_id
+     *   content:      "join"                   "john"
+     *   spanList:     (0, 4, "join", content)  (0, 4, "john", content)
```

Comment:
Since this class is named `join`, try to us a different example string pair, such as `schwarzenegger` versus `schwarzeneger`